src/code/containers/challenge-container-selector.js

```diff
@@ -43,27 +44,33 @@ class ChallengeContainerSelector extends Component {
   }
 
   componentWillMount() {
-    const { routeParams, navigateToCurrentRoute, navigateToChallenge } = this.props;
-    if (hasChangedRouteParams(this.props)) {
+    const { navigateToCurrentRoute, navigateToChallenge, authoring } = this.props;
+    // the URL's challengeId is only used for initial routing, so prioritize the numeric route params
```

Comment:
This is the main part I have questions about- I have numerical paths take priority over written challenge IDs because it required the least effort on my part to update the URL. Does that seem reasonable for now?